homeassistant/components/camera/__init__.py

```diff
@@ -500,3 +547,31 @@ def _write_image(to_file, image_data):
             _write_image, snapshot_file, image)
     except OSError as err:
         _LOGGER.error("Can't write image to file: %s", err)
+
+
+async def async_handle_play_stream_service(camera, service):
+    """Handle play stream services calls."""
+    from homeassistant.components.media_player.const import (
+        ATTR_MEDIA_CONTENT_ID, ATTR_MEDIA_CONTENT_TYPE,
+        SERVICE_PLAY_MEDIA, DOMAIN as DOMAIN_MP)
+    from homeassistant.components.stream import request_stream
+
+    if not camera.stream_source:
+        raise HomeAssistantError("{} does not support play stream service"
+                                 .format(camera.entity_id))
+
+    hass = camera.hass
+    fmt = service.data[ATTR_FORMAT]
+    entity_ids = service.data[ATTR_MEDIA_PLAYER]
+
+    url = request_stream(hass, camera.stream_source, fmt=fmt)
+    data = {
+        ATTR_MEDIA_CONTENT_ID: url,
+        ATTR_MEDIA_CONTENT_TYPE: 'application/vnd.apple.mpegurl'
```

Comment:
Doesn't this depend on the value of `fmt` ? We should also have the format <-> content type mapping live inside the stream component.